web/modules/custom/dept_publications/src/Routing/SecurePublicationsRouteSubscriber.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace Drupal\dept_publications\Routing;
+
+use Drupal\Core\Routing\RouteSubscriberBase;
+use Symfony\Component\Routing\RouteCollection;
+
+/**
+ * Listens to the dynamic route events.
+ */
+class SecurePublicationsRouteSubscriber extends RouteSubscriberBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function alterRoutes(RouteCollection $collection) {
+    if ($route = $collection->get('origins_workflow.moderation_state_controller_change_state')) {
+      $route->setRequirements([
+        '_custom_access' => '\Drupal\dept_publications\Controller\SecurePublicationsPermissionController::accessCheckModerationStateChange',
```

Comment:
YOINK!

We step in and hijack the existing controller definition. Here we can then evaluate our own requirements and, if all else fails, proxy on the request to the existing origins workflow controller and let it continue to do its thing.